### PR TITLE
Add a fun new preview text in the SUI, enable the cursor

### PR DESCRIPTION
**Pull request by DHowett**
Our existing preview text was not very helpful in learning how different settings impacted the display of text in Terminal.

This new preview text contains:
* Bold text, which is controlled by intenseTextStyle
* Colors
* Emoji
* A cursor, which overlaps a single character to show inversion behavior

### CODE REVIEW

**Comment by DHowett** [comment]
preview1
preview2

**Comment by DHowett** [comment]
I have something even MORE fun, but i was holding it for after this...

File: src/cascadia/TerminalControl/TermControl.idl [hunk 1]
```diff
@@ -111,6 +111,8 @@ namespace Microsoft.Terminal.Control
         // opacity set by the settings should call this instead.
         Double BackgroundOpacity { get; };
 
+        Boolean DisplayCursorWhileBlurred;
```
Comment (DHowett) [inline src/cascadia/TerminalControl/TermControl.idl:114]: blur is the opposite of focus, don't at me

File: src/cascadia/TerminalControl/TermControl.cpp [hunk 2]
```diff
@@ -997,7 +997,11 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             // _cursorTimer doesn't exist, and it would never turn on the
             // cursor. To mitigate, we'll initialize the cursor's 'on' state
             // with `_focused` here.
-            _core.CursorOn(_focused);
+            _core.CursorOn(_focused || DisplayCursorWhileBlurred);
+            if (DisplayCursorWhileBlurred)
+            {
+                _cursorTimer->Start();
+            }
```
Comment (lhecker) [inline src/cascadia/TerminalControl/TermControl.cpp:1004]: This doesn't work if the `blinkTime` is `INFINITE`?

**Review by lhecker — APPROVED** [review]
I'm not happy about the `DisplayCursorWhileBlurred` because it's a hack. It would probably make more sense if it was part of the unfocused appearance as a proper setting. It might be worthwhile to note in the .idl that the setting is bodgy/hacky.

File: src/cascadia/TerminalControl/TermControl.cpp (hunk 2 above)
Comment (DHowett) [reply src/cascadia/TerminalControl/TermControl.cpp:1004]: nope, but that's not new from this code.

**Comment by DHowett** [comment]
> I'm not happy about the DisplayCursorWhileBlurred because it's a hack. 

I agree, but I did look at other XAML properties like `AllowFocusWhenDisabled` and stuff. There is precedent for these weird single-shot p\roperties.

I do think we should fix Unfocused Appearances, but in general doing that will utterly F up the settings UI preview because it is *always* unfocused... SO WE WILL STILL NEED A HACK :grin:

**Comment by lhecker** [comment]
> I do think we should fix Unfocused Appearances, but in general doing that will utterly F up the settings UI preview because it is always unfocused... SO WE WILL STILL NEED A HACK 😁

We could technically just separate out the WinUI and non-WinUI parts from the `_GotFocusHandler` and `_LostFocusHandler` and expose the latter as a property. This would allow you to switch between the two appearances at will. I do agree though that we can do something like this at a later time (unless you feel like this would be easy to implement?). As you mentioned, there's a lot of subjective design issues in the area already that we'd probably need to address.

**Review by PankajBhojwani — APPROVED** [review]

**Pull request merged**